stdlib/socket/0/socket.rbs

```diff
@@ -0,0 +1,2687 @@
+  # *recvfrom_nonblock* fails.
+  #
+  # Socket#recvfrom_nonblock may raise any error corresponding to recvfrom(2)
+  # failure, including Errno::EWOULDBLOCK.
+  #
+  # If the exception is Errno::EWOULDBLOCK or Errno::EAGAIN, it is extended by
+  # IO::WaitReadable. So IO::WaitReadable can be used to rescue the exceptions for
+  # retrying recvfrom_nonblock.
+  #
+  # By specifying a keyword argument *exception* to `false`, you can indicate that
+  # recvfrom_nonblock should not raise an IO::WaitReadable exception, but return
+  # the symbol `:wait_readable` instead.
+  #
+  # ### See
+  # *   Socket#recvfrom
+  #
+  #
+  def recvfrom_nonblock: (Integer maxlen, ?Integer flags, ?untyped outbuf, ?exception: boolish) -> ([String, Addrinfo] | :wait_readable)
+
+  # Accepts an incoming connection returning an array containing the (integer)
+  # file descriptor for the incoming connection, *client_socket_fd*, and an
+  # Addrinfo, *client_addrinfo*.
+  #
+  # ### Example
+  #     # In one script, start this first
+  #     require 'socket'
+  #     include Socket::Constants
+  #     socket = Socket.new( AF_INET, SOCK_STREAM, 0 )
+  #     sockaddr = Socket.pack_sockaddr_in( 2200, 'localhost' )
+  #     socket.bind( sockaddr )
+  #     socket.listen( 5 )
+  #     client_fd, client_addrinfo = socket.sysaccept
+  #     client_socket = Socket.for_fd( client_fd )
+  #     puts "The client said, '#{client_socket.readline.chomp}'"
+  #     client_socket.puts "Hello from script one!"
+  #     socket.close
+  #
+  #     # In another script, start this second
+  #     require 'socket'
+  #     include Socket::Constants
+  #     socket = Socket.new( AF_INET, SOCK_STREAM, 0 )
+  #     sockaddr = Socket.pack_sockaddr_in( 2200, 'localhost' )
+  #     socket.connect( sockaddr )
+  #     socket.puts "Hello from script 2."
+  #     puts "The server said, '#{socket.readline.chomp}'"
+  #     socket.close
+  #
+  # Refer to Socket#accept for the exceptions that may be thrown if the call to
+  # *sysaccept* fails.
+  #
+  # ### See
+  # *   Socket#accept
+  #
+  #
+  def sysaccept: () -> [Integer, Addrinfo]
+
+  private
+
+  def __accept_nonblock: (untyped) -> untyped
+
+  def __connect_nonblock: (untyped, untyped) -> untyped
+
+  def __recvfrom_nonblock: (untyped, untyped, untyped, untyped) -> untyped
+
+  # Creates a new socket object.
+  #
+  # *domain* should be a communications domain such as: :INET, :INET6, :UNIX, etc.
+  #
+  # *socktype* should be a socket type such as: :STREAM, :DGRAM, :RAW, etc.
+  #
+  # *protocol* is optional and should be a protocol defined in the domain. If
+  # protocol is not given, 0 is used internally.
+  #
+  #     Socket.new(:INET, :STREAM) # TCP socket
+  #     Socket.new(:INET, :DGRAM)  # UDP socket
+  #     Socket.new(:UNIX, :STREAM) # UNIX stream socket
+  #     Socket.new(:UNIX, :DGRAM)  # UNIX datagram socket
+  #
+  def initialize: (Symbol domain, Symbol socktype, ?Integer protocol) -> untyped
```

Comment:
It seems we can write `socket = Socket.new(Socket::AF_INET, Socket::SOCK_STREAM, 0)` where the parameters are `Integer`?